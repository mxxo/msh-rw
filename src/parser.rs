use crate::*;
use std::path::PathBuf;
use nom::*;
use thiserror::Error;

#[derive(Error, Debug)]
pub enum MshError {
    #[error("invalid header (expected {expected:?}, found {found:?})")]
    InvalidHeader {
        expected: String,
        found: String,
    },
}

pub type MshResult<T> = std::result::Result<T, MshError>;

impl Msh2 {
    fn from_file(file: &PathBuf) -> MshResult<Msh2> {
        todo!();
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    #[test]
    fn empty_mesh() {
        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
        path.push("props");
        path.push("empty.msh");
        Msh2::from_file(&path).unwrap();
    }
}
