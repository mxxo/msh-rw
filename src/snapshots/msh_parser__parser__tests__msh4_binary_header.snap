---
source: src/parser.rs
expression: check_header(&path).unwrap()
---
MshHeader {
    version: V41,
    storage: BinaryLe,
    size_t: EightBytes,
}
